### PR TITLE
[PRODDEV-393] Use first name instead of username

### PR DESCRIPTION
**Pull request by imankulov**
### Description

TASK: [PRODDEV-393](https://edraak.atlassian.net/browse/PRODDEV-393)

- Add a helper pseudo-property UserProfile.first_name.
- Update header templates.

### Notes

**Important:** I coun't create a new legacy course and made changes blindly.

### Related PRs

- https://github.com/Edraak/marketing-site/pull/673
- https://github.com/Edraak/edraak-programs/pull/2469

### CODE REVIEW

**Review by qutaiba — APPROVED** [review]

**Review by umar221b — APPROVED** [review]

File: lms/djangoapps/django_comment_client/utils.py [hunk 1]
```diff
@@ -769,6 +769,7 @@ def prepare_content(content, course_key, is_staff=False, discussion_division_ena
                 ("username" in fields or has_permission(endorser, "endorse_comment", course_key))
         ):
             endorsement["username"] = endorser.username
+            endorsement["user_first_name"] = endorser.profile.name
```
Comment (qutaiba) [inline lms/djangoapps/django_comment_client/utils.py]: Shouldn't this be `endorser.profile.first_name`?

**Review by qutaiba — CHANGES_REQUESTED** [review]

**Comment by qutaiba** [comment]
Hi @imankulov our PM Omar added a new ticket to show full name under [forums](https://edraak.atlassian.net/browse/PRODDEV-463) it makes sense to commit it on this branch

File: lms/lib/comment_client/utils.py [hunk 2]
```diff
@@ -192,3 +196,60 @@ def check_forum_heartbeat():
...
+    cache_timeout = 3600
+    cache_key = 'comment_client.get_first_name.{}'.format(user_id)
+    first_name = cache.get(cache_key)
+    if first_name is user_not_found:
+        return None
+
+    if first_name is None:
+        try:
+            first_name = User.objects.get(id=user_id).profile.first_name
+        except User.NotFound:
```
Comment (qutaiba) [inline lms/lib/comment_client/utils.py]:
```suggestion
        except User.DoesNotExist:
```

Comment (qutaiba) [inline lms/lib/comment_client/utils.py]: This is causing Jenkins errors 
`[](http://build.edraak.cc/blue/organizations/jenkins/edraak-platform/detail/PR-263/6/pipeline/#step-256-log-1427)[](http://build.edraak.cc/blue/organizations/jenkins/edraak-platform/detail/PR-263/6/pipeline/#step-256-log-1428)File "/home/jenkins/workspace/edraak-platform_PR-263/lms/lib/comment_client/utils.py", line 251, in get_first_name

[2022-03-18T18:30:16.251Z]     except User.NotFound:

[2022-03-18T18:30:16.251Z] AttributeError: type object 'User' has no attribute 'NotFound'`

**Review by qutaiba — CHANGES_REQUESTED** [review]

**Comment by qutaiba** [comment]
@a-nassif your review here is needed

File: lms/lib/comment_client/utils.py (hunk 2 above)
Comment (imankulov) [reply lms/lib/comment_client/utils.py]: Oh, thanks for the catch 😬

File: lms/djangoapps/django_comment_client/utils.py (hunk 1 above)
Comment (imankulov) [reply lms/djangoapps/django_comment_client/utils.py]: Great catch, thanks! Since we decided to show the full name, I'll rewrite this part anyway.

**Comment by imankulov** [comment]
Fixed broken tests 🎉

**Review by qutaiba — APPROVED** [review]

File: lms/templates/user_dropdown.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
 <%
 ## This template should not use the target student's details when masquerading, see TNL-4895
 self.real_user = getattr(user, 'real_user', user)
-username = get_enterprise_learner_generic_name(request) or self.real_user.username
+username = get_enterprise_learner_generic_name(request) or self.real_user.profile.name
```
Comment (qutaiba) [inline lms/templates/user_dropdown.html]: @imankulov IF this is related to the header and not the discussion it should be user first name not full name

**Review by qutaiba — CHANGES_REQUESTED** [review]

File: lms/templates/user_dropdown.html (hunk 3 above)
Comment (imankulov) [reply lms/templates/user_dropdown.html]: Great catch, thanks! Addressed in 4f5907c12b3c8c48dcee9fb23ee761e8aaeeae15

**Review by qutaiba — APPROVED** [review]

**Review by qutaiba — APPROVED** [review]

**Pull request merged**